app/edge/devices/components/EdgeDevicesDatatable/EdgeDevicesDatatableActions.tsx

```diff
@@ -0,0 +1,125 @@
+import { useRouter } from '@uirouter/react';
+
+import type { Environment } from '@/portainer/environments/types';
+import { Button } from '@/portainer/components/Button';
+import { confirm } from '@/portainer/services/modal.service/confirm';
+import { prompt } from '@/portainer/services/modal.service/prompt';
+import * as notifications from '@/portainer/services/notifications';
+import { activateDevice } from '@/portainer/hostmanagement/open-amt/open-amt.service';
+
+interface Props {
+  selectedItems: Environment[];
+  isFDOEnabled: boolean;
+  isOpenAMTEnabled: boolean;
+  setLoadingMessage(message: string): void;
+}
+
+export function EdgeDevicesDatatableActions({
+  selectedItems,
+  isOpenAMTEnabled,
+  isFDOEnabled,
+  setLoadingMessage,
+}: Props) {
+  const hasSelectedItem = selectedItems.length === 1;
+
+  const router = useRouter();
+
+  return (
+    <div className="actionBar">
+      {isFDOEnabled && (
+        <Button onClick={() => onAddNewDeviceClick()}>
+          <i className="fa fa-plus-circle space-right" aria-hidden="true" />
+          Add new
+        </Button>
+      )}
+
+      {isOpenAMTEnabled && (
+        <Button
+          disabled={!hasSelectedItem}
+          onClick={() => onAssociateOpenAMTClick(selectedItems)}
+        >
+          <i className="fa fa-link space-right" aria-hidden="true" />
+          Associate with OpenAMT
+        </Button>
+      )}
+    </div>
+  );
+
+  function onAddNewDeviceClick() {
+    prompt({
```

Comment:
we don't have promptAsync, but it should be easy to create one:

```
async function promptAsync(options: Omit<PromptOptions, 'callback'>) {
  return new Promise((callback) => prompt({...options, callback}))
}

Reply:
done